ledger/participant-integration-api/src/main/scala/platform/store/backend/common/EventStorageBackendTemplate.scala

```diff
@@ -485,6 +485,104 @@ abstract class EventStorageBackendTemplate(
   )(connection: Connection, loggingContext: LoggingContext): Unit = {
     import com.daml.platform.store.backend.Conversions.OffsetToStatement
 
+    def unionWithSelectImmediateDivulgenceCreatedEvents(): ComposableQuery.CompositeSql =
+      if (pruneAllDivulgedContracts) {
+        val pruneAfterClause = {
+          // We need to distinguish between the two cases since lexicographical comparison
+          // in Oracle doesn't work with '' (empty strings are treated as NULLs) as one of the operands
+          participantAllDivulgedContractsPrunedUpToInclusive(connection) match {
+            case Some(pruneAfter) => cSQL"and event_offset > $pruneAfter"
+            case None => cSQL""
+          }
+        }
+
+        cSQL"""
+            -- Immediate divulgence pruning
+            UNION
+            SELECT contract_id, event_sequential_id, transaction_id
+              FROM participant_events_create c, parameters p
+             WHERE
+             (
+               event_offset > p.participant_all_divulged_contracts_pruned_up_to_inclusive -- TODO pruning check
+               OR p.participant_all_divulged_contracts_pruned_up_to_inclusive IS NULL
+             )
+             AND event_offset <= $pruneUpToInclusive
+             -- Only prune create events which did not have a locally hosted party before their creation offset
+             AND NOT EXISTS (
+              SELECT 1
+                FROM party_entries p
+               WHERE p.typ = 'accept'
+                 AND p.ledger_offset <= c.event_offset
+                 AND #${queryStrategy.isTrue("p.is_local")}
+                 AND #${queryStrategy.arrayContains("c.flat_event_witnesses", "p.party_id")}
+             )
+             $pruneAfterClause
+         """
+      } else cSQL""""""
+
+    pruneWithLogging("prune all")(SQL"""
+       WITH archived_events AS (
+         SELECT contract_id, event_sequential_id
+           FROM participant_events_consuming_exercise
+          WHERE event_offset <= $pruneUpToInclusive
+       ),
+       created_events_to_be_deleted AS (
+          SELECT contract_id, event_sequential_id, transaction_id FROM participant_events_create creates_to_be_deleted
+                WHERE creates_to_be_deleted.contract_id IN (SELECT contract_id FROM archived_events)
+          ${unionWithSelectImmediateDivulgenceCreatedEvents()} -- TODO pruning: Merge in where clause
+       ),
+       creates_distinct_transaction_ids_to_be_pruned AS (
+         -- TODO pruning: Might not be needed
+         SELECT DISTINCT(transaction_id) AS transaction_id FROM created_events_to_be_deleted
+       ),
+       nonconsuming_events_to_be_deleted AS (
+          SELECT contract_id, event_sequential_id FROM participant_events_non_consuming_exercise nonconsuming_to_be_deleted
+            WHERE nonconsuming_to_be_deleted.contract_id IN (SELECT contract_id FROM archived_events)
```

Comment:
I think the status quo is that all non-consuming events within the pruning range are being deleted, so contract_id check shouldn't be here?

Reply:
even non consuming events for non-archived contracts?

Reply:
Yes it seems this is the status quo, which is odd to be honest since we are chunking events from potential active contracts as well only to feel safe because our tx tree stream restriction doesn't allow subscribing before the pruning offset. Anyway, good point, reverting